### ISSUE TITLE
Creating StaticSiteCustomDomain waits for the TXT validation indefinitely

### ISSUE THREAD

**Issue by rmaziarka** [pulumi/pulumi-azure-native#2296]
### What happened?

When you create StaticSiteCustomDomain for StaticSite it waits until TXT validation is finished. But it won't be finished, because TXT entry is being created in the next step. Scripts wait for the execution indefinitely.

### Expected Behavior

Creation of StaticSiteCustomDomain shouldn't wait for the domain validation, and it should proceed without it.

### Steps to reproduce

```csharp
    var staticSiteCustomDomain = new StaticSiteCustomDomain(siteName, new()
    {
        DomainName = "page.com",
        Name = staticSite.Name,
        ResourceGroupName = resourceGroup.Name,
        ValidationMethod = "dns-txt-token",
    });

    var txtRecordSet = new RecordSet(siteName+"-TXT", new()
    {
        RecordType = "TXT",
        RelativeRecordSetName = "",
        ResourceGroupName = resourceGroup.Name,,
        TargetResource = new Pulumi.AzureNative.Network.Inputs.SubResourceArgs
        {
            Id = staticSite.Id,
        },
        TxtRecords = new[]
        {
            new TxtRecordArgs
            {
                Value = new[]
                {
                    staticSiteCustomDomain.ValidationToken
                },
            },
        },
        Ttl = 3600,
        ZoneName = "page.com",
    });
```

### Output of `pulumi about`

CLI
Version      3.57.1
Go Version   go1.20.1
Go Compiler  gc

Plugins
NAME          VERSION
azure-native  1.94.0
dotnet        unknown

Host
OS       Microsoft Windows 11 Pro
Version  10.0.22621 Build 22621
Arch     x86_64

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

**Comment by danielrbradley** [comment]
It seems [the Azure CLI has the same default behaviour](https://learn.microsoft.com/en-us/cli/azure/staticwebapp/hostname?view=azure-cli-latest#az-staticwebapp-hostname-set) and has an extra `--no-wait` argument to change this. As a workaround, you could invoke this via the CLI using [the command provider](https://www.pulumi.com/registry/packages/command/).

**Comment by rmaziarka** [comment]
Thanks @danielrbradley for the quick look. Right now I just commented StaticSiteCustomDomain and created it manually, but in the long run, it would be great to have this option in Pulumi.

**Comment by danielrbradley** [comment]
Yup, we'll keep this on the backlog, but would also welcome PRs if anyone wants to look into this further before we get to it.

**Comment by icgarzon** [comment]
I just got this problem as well. Do you guys know if there’s another solution, or is it definitely not usable?
